app/context_manager.py

```diff
@@ -0,0 +1,25 @@
+from os import remove, makedirs, path
+from types import TracebackType
+from typing import Optional, Type
+
+
+class MoveFileManager:
+    def __init__(
+            self,
+            source_file_name: str,
+            path: str
+    ) -> None:
+        self.source_file_name = source_file_name
+        self.path = path
+
+    def __enter__(self) -> None:
+        if self.path and not path.exists(self.path):
+            makedirs(self.path)
+
+    def __exit__(
+            self,
+            exc_type: Optional[Type[BaseException]],
+            exc_val: Optional[Type[BaseException]],
+            exc_tb: Optional[TracebackType]
+    ) -> None:
+        remove(self.source_file_name)

```

Comment:
The __exit__ method should handle exceptions properly. Consider checking if exc_type is not None and handling the exception accordingly, or re-raising it if necessary.